.github/workflows/cacti-build-and-test.yaml

```diff
@@ -0,0 +1,54 @@
+# Unified Cacti Build and Test
+---
+env:
+  NODEJS_VERSION: v18.18.2
+  
+jobs:
+
+  build-dev:
+    continue-on-error: false
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4.1.1
+      - uses: ./.github/actions/build-and-cache/
+
+  yarn-tasks:
+    name: yarn ${{ matrix.yarn-task }}
+    needs: build-dev
+    runs-on: ubuntu-22.04
+    continue-on-error: true
+    strategy:
+      matrix:
+        yarn-task: 
+        - tools:validate-bundle-names
+        - custom-checks
+        - lint
+    steps:
+      - uses: actions/checkout@v4.1.1
+      - uses: ./.github/actions/build-and-cache/
+      - run: yarn ${{ matrix.yarn-task }}
+      
+  package-test:
+    name: test ${{ matrix.package-to-test }}
+    needs: build-dev
+    runs-on: ubuntu-22.04
+    continue-on-error: true
+    strategy:
+      matrix:
+        # this can eventually be the output of npx 'lerna changed' or 'lerna list', depending on how invoked
+        package-to-test: 
+        - cactus-api-client
+        - cactus-verifier-client
+        - cactus-test-plugin-consortium-manual
+    steps:
+      - uses: actions/checkout@v1
```

Comment:
@jenniferlianne Is there any reason to deviate from the version of the checkout action we are using throughout the CI?

Reply:
@jenniferlianne One more question, do we need this checkout if the cacti-build-and-test.yaml file also does a checkout?

Reply:
updated.